packages/babel-helper-create-regexp-features-plugin/src/util.js

```diff
@@ -0,0 +1,48 @@
+import { FEATURES, hasFeature } from "./features";
+
+export function generateRegexpuOptions(node, features) {
+  let useUnicodeFlag = false,
+    dotAllFlag = false,
+    unicodePropertyEscape = false,
+    namedGroup = false;
+  const { flags, pattern } = node;
+  const flagsIncludesU = flags.includes("u");
+
+  if (flagsIncludesU === true) {
+    if (hasFeature(features, FEATURES.unicodeFlag) === false) {
```

Comment:
Nit: Don't you like `!`? :stuck_out_tongue: 
Also all the `=== true`s are unnecessary.

Reply:
Actually I have no preference on `!` or `=== false` until I read the compiled instructions:

This is what happens under the hood when we write `if(a) { // blablabla }`

```asm
0x1fafde0d1bf2    32  4c8b4510       REX.W movq r8,[rbp+0x10]
0x1fafde0d1bf6    36  41f6c001       testb r8,0x1
0x1fafde0d1bfa    3a  0f84cf140000   jz 0x1fafde0d30cf  <+0x150f>
                  -- B4 start --
0x1fafde0d1c00    40  4d3945f8       REX.W cmpq [r13-0x8] (root (false_value)),r8
0x1fafde0d1c04    44  0f8438000000   jz 0x1fafde0d1c42  <+0x82>
                  -- B5 start --
0x1fafde0d1c0a    4a  4d394500       REX.W cmpq [r13+0x0] (root (empty_string)),r8
0x1fafde0d1c0e    4e  0f842e000000   jz 0x1fafde0d1c42  <+0x82>
                  -- B6 start --
0x1fafde0d1c14    54  4d8b48ff       REX.W movq r9,[r8-0x1]
0x1fafde0d1c18    58  41f6410d10     testb [r9+0xd],0x10
0x1fafde0d1c1d    5d  0f851f000000   jnz 0x1fafde0d1c42  <+0x82>
                  -- B7 start --
                  -- B8 start --
0x1fafde0d1c23    63  4d398d80000000 REX.W cmpq [r13+0x80] (root (heap_number_map)),r9
0x1fafde0d1c2a    6a  0f8486140000   jz 0x1fafde0d30b6  <+0x14f6>
                  -- B9 start --
0x1fafde0d1c30    70  4d398d40010000 REX.W cmpq [r13+0x140] (root (bigint_map)),r9
0x1fafde0d1c37    77  0f8466140000   jz 0x1fafde0d30a3  <+0x14e3>
0x1fafde0d1c3d    7d  e913000000     jmp 0x1fafde0d1c55  <+0x95>
```
Basically it compares `a` to these four objects: `false`, `""`, `0` and `0n`. Since JavaScript is weak-typed, even if you know for sure `a` could only be a boolean, the compiler would not simplify `if(a)` to the following codes, which is simplified from `if(a === true)`.

```asm
0x1fafde0d1bf2    32  4c8b4510       REX.W movq r8,[rbp+0x10]
0x1fafde0d1bf6    36  41f6c001       testb r8,0x1
0x1fafde0d1bfa    3a  0f84cf140000   jz 0x1fafde0d30cf  <+0x150f>
                  -- B4 start --
0x1fafde0d1c00    40  4d3945f8       REX.W cmpq [r13-0x8] (root (false_value)),r8
0x1fafde0d1c04    44  0f8438000000   jz 0x1fafde0d1c42  <+0x82>
```

However, these branch instructions would never be our performance bottleneck thanks to complex branch prediction in our modern CPU. I am happy to rewrite into more common code style as long as you find out. 😀